File: Windows/generate-attacks-list.py
from config import WORDLISTS, DICTIONARIES, RULES, MASKS

def generate_attack_list(attack_type):
    attack_list = []

    for wordlist in WORDLISTS:
        for rule in RULES:
            attack_list.append(["-a 0", wordlist, rule])

    for dictionary in DICTIONARIES:
        for rule in RULES:
            attack_list.append(["-a 0", dictionary, rule])

    for rule in RULES:
        attack_list.append(["-a 0", "", rule])

    for mask in MASKS:
        attack_list.append(["-a 3", mask])

    for wordlist in WORDLISTS:
        for mask in MASKS:
            attack_list.append(["-a 6", wordlist, mask])

    for dictionary in DICTIONARIES:
        for mask in MASKS:
            attack_list.append(["-a 6", dictionary, mask])

    return attack_list

def write_attacks_to_file(attacks):
    with open("attacks_list.py", "w") as file:
        file.write("# attacks_list.py\n")
        file.write("# Generated attacks list\n\n")
        file.write("attacks = [\n")
        for attack in attacks:
            file.write(f"    {attack},\n")
        file.write("]\n")

if __name__ == "__main__":
    attack_type = 0  # Set the desired attack type
    attacks = generate_attack_list(attack_type)
    
    write_attacks_to_file(attacks)

File: OS X/scripts/generate-attacks-list.js
const { WORDLISTS, DICTIONARIES, RULES, MASKS } = require('../config');

const generateAttackList = (attackType) => {
	const attackList = [];

	for (const wordlist of WORDLISTS) {
		for (const rule of RULES) {
			attackList.push(["-a 0", wordlist, rule]);
		}
	}

	for (const dictionary of DICTIONARIES) {
		for (const rule of RULES) {
			attackList.push(["-a 0", dictionary, rule]);
		}
	}

	for (const rule of RULES) {
		attackList.push(["-a 0", "", rule]);
	}

	for (const mask of MASKS) {
		attackList.push(["-a 3", mask]);
	}

	for (const wordlist of WORDLISTS) {
		for (const mask of MASKS) {
			attackList.push(["-a 6", wordlist, mask]);
		}
	}

	for (const dictionary of DICTIONARIES) {
		for (const mask of MASKS) {
			attackList.push(["-a 6", dictionary, mask]);
		}
	}

	return attackList;
};

const writeAttacksToFile = (attacks) => {
	const fs = require('fs');
	const path = require('path');
	const filePath = path.join(__dirname, 'attacks_list.js');

	const content = `
// attacks_list.js
// Generated attacks list

const attacks = ${JSON.stringify(attacks, null, 2)};

module.exports = attacks;
`;

	fs.writeFileSync(filePath, content);
	console.log(`Attacks list written to ${filePath}`);
};

if (require.main === module) {
	const attackType = 0;  // Set the desired attack type
	const attacks = generateAttackList(attackType);
  
	writeAttacksToFile(attacks);
}
